SteamcLog/Classes/SteamcLog.swift

```diff
@@ -104,6 +104,23 @@ public struct SteamcLog {
         return urls[urls.endIndex - 1].appendingPathComponent("steamclog.txt")
     }()
 
+    private let previousLogFilePath: URL? = {
+        // get a list of cache directories
+        let urls = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask)
+
+        // grab the set of archived logs from the cache directory
+        let cacheDir = urls[urls.endIndex - 1]
```

Comment:
Do we need to worry about urls being empty here and causing a crash in some weird instances?

Reply:
I don't think so. That's the OS created application cache directory, so in theory it should always exist, and though the logic is a little weird (I don't fully understand what multiple paths there mean and why it is getting the last one, and `urls.last!` seems like it would be clearer), it is the same logic used in the code to get the current log file path in steamclog and inside XCGLogger itself if it generates that path (though it puts it in a subdirectory), so if it were dangerous in some at all possible edge case, it would be crashing in the wild right now.

I think what I WILL do, is separate the code to get the cache directory out of those two functions and maybe add a comment about the strangeness of it.